UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Views/NoticestoMariners/ShowWeeklyFilesList.cshtml

```diff
@@ -1,30 +1,84 @@
 ﻿@using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners
 @model List<ShowFilesResponseModel>
 @{
-    var a = Model;
+    bool publicBool = false;
+    bool partnerBool = false;
 }
-    <table class="table table-scroll small-first-col notice-to-mariners-weekly-table">
-        <caption class="hidden-caption">Notices to Mariners Weekly Updates</caption>
-        <thead>
-            <tr>
-                <th>File Name</th>
-                <th>File Size</th>
-                <th class="th-3"><span class="visually-hidden">Action</span></th>
-            </tr>
-        </thead>
-        <tbody class="scroll-content scroll-bar">
-            @for (int i = 0; i < Model.Count; i++)
+
+    @if (ViewBag.IsDistributor)
+    {
+        <caption class="table-caption">Partner</caption>
+        <table class="table table-scroll small-first-col notice-to-mariners-weekly-table">
+            <caption class="hidden-caption">Notices to Mariners Weekly Updates</caption>
```

Comment:
Partner & Public table code is duplicate. Please refactor to remove duplicate code.

Reply:
Refactor done